academic-observatory-workflows/academic_observatory_workflows/openalex_telescope/telescope.py

```diff
@@ -0,0 +1,502 @@
+# Copyright 2022-2024 Curtin University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Aniek Roelofs, James Diprose, Alex Massen-Hane
+
+from __future__ import annotations
+
+import logging
+from typing import List, Optional
+
+import pendulum
+from airflow import DAG
+from airflow.decorators import dag, task, task_group
+from airflow.operators.empty import EmptyOperator
+
+from academic_observatory_workflows.config import project_path
+from academic_observatory_workflows.openalex_telescope.release import OpenAlexEntity
+from observatory_platform.airflow.airflow import is_first_dag_run, on_failure_callback
+from observatory_platform.airflow.release import release_to_bucket
+from observatory_platform.airflow.sensors import PreviousDagRunSensor
+from observatory_platform.airflow.tasks import check_dependencies, gke_create_storage, gke_delete_storage
+from observatory_platform.airflow.workflow import cleanup, CloudWorkspace, make_workflow_folder
+from observatory_platform.config import AirflowConns
+from observatory_platform.google.bigquery import bq_create_dataset
+from observatory_platform.google.gke import DEFAULT_GKE_IMAGE, GkeParams
+
+
+class DagParams:
+    def __init__(
+        self,
+        *,
+        dag_id: str,
+        cloud_workspace: CloudWorkspace,
+        bq_dataset_id: str = "openalex",
+        api_bq_dataset_id: str = "dataset_api",
+        entity_names: List[str] = None,
+        schema_folder: str = project_path("openalex_telescope", "schema"),
+        dataset_description: str = "The OpenAlex dataset: https://docs.openalex.org/",
+        non_concurrent_table_expiry_days: int = 62,
+        n_transfer_trys: int = 3,
+        primary_key: str = "id",
+        aws_conn_id: str = "aws_openalex",
+        aws_openalex_bucket: str = "openalex",
+        slack_conn_id: Optional[str] = AirflowConns.SLACK,
+        start_date: pendulum.DateTime = pendulum.datetime(2021, 12, 1),
+        schedule: str = "@weekly",
+        queue: str = "remote_queue",
```

Comment:
Is the remote queue still relevant? I think we can ditch it

Reply:
Yeah you're right, I've removed it.